### PR TITLE
ReceiptTrigger+Tests+DataFactory+DoctorTrigger

### PR DESCRIPTION
**Pull request by sergeybaklanov**

### CODE REVIEW

File: force-app/main/default/classes/ReceiptTriggerHandler.cls [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    private static void updateTreatmentPaymentsAmount(Map<Id, Receipt__c> newReceiptsMap){
+        List<Case> treatmentsList = [SELECT Id, (SELECT Id
+                                                 FROM Receipts__r
+                                                 WHERE Id IN:newReceiptsMap.keySet())
+                                     FROM Case ];
+        List<Id> treatmentIdsList = new List<Id>();
+        for (Case treatment_i : treatmentsList){
+            treatmentIdsList.add(treatment_i.Id);
```
Comment (egorsolotin) [inline force-app/main/default/classes/ReceiptTriggerHandler.cls]: Better to make map and take keySet() from it to obtain the list of Ids. This solution is recommended by Salesforce

File: force-app/main/default/classes/ReceiptTriggerHandler.cls [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        List<AggregateResult> treatmentReceiptsList = [SELECT SUM(Amount__c)Amount, Treatment__c TrId
+                                                       FROM Receipt__c 
+                                                       WHERE Treatment__c IN:treatmentIdsList
+                                                       GROUP BY Treatment__c];
+
+        List<Case> treatmentsToUpdateList = new List<Case>();
+        for (AggregateResult treatment_i : treatmentReceiptsList){
+            treatmentsToUpdateList.add(new Case(Id = (Id) treatment_i.get('TrId'), Amount_of_payments__c = (Decimal) treatment_i.get('Amount')));
+        }
+        List<Database.SaveResult> updateResults = Database.update (treatmentsToUpdateList, true);
```
Comment (egorsolotin) [inline force-app/main/default/classes/ReceiptTriggerHandler.cls:74]: You don't use this var

File: force-app/main/default/classes/ReceiptTriggerHandler.cls [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                                                     WHERE Best_doctor_of_the_month__c = true
+                                                     ORDER BY Payments_amount__c DESC)
+                                         FROM Account 
+                                         WHERE Id IN:doctorsDepartmentsIdsList];
+
+        List<Contact> finalDoctorsUpdateList = new List<Contact>();
+
+        for (Account department_i : departmentsList){
+            for (Integer i = 0; i < department_i.Contacts.size(); i++){
+                if (i == 0){
```
Comment (egorsolotin) [inline force-app/main/default/classes/ReceiptTriggerHandler.cls]: Why you don't start for loop from i==1?

File: force-app/main/default/classes/ReceiptTriggerHandlerTest.cls [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        List<Contact> doctors = TestDataFactory.createDoctorsList(1, departments, true);
+        List<InsuranceCompany__c> insuranceCompanies = TestDataFactory.createInsuranceCompanyList(1, true);
+        List<Patient__c> patients = TestDataFactory.createPatientList(1, hospitals, true);
+        List<Insurance__c> insurance = TestDataFactory.createInsuranceList(1, 150, insuranceCompanies, patients, true);
+        List<Case> treatments = TestDataFactory.createTreatmentList(1, departments, doctors, patients, insurance, true);
+    }
+
+    @isTest
+    static void testUpdateDoctorsPaymentsAmount(){
```
Comment (egorsolotin) [inline force-app/main/default/classes/ReceiptTriggerHandlerTest.cls]: Test method should have Asserts

**Review by egorsolotin — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/ReceiptTriggerHandler.cls (hunk 2 above)
Comment (sergeybaklanov) [reply force-app/main/default/classes/ReceiptTriggerHandler.cls:74]: I use it in the method  dmlStatusNotification(update results) in the following line

**Review by egorsolotin — APPROVED** [review]

**Pull request merged**